scipy/signal/signaltools.py

```diff
@@ -56,25 +56,19 @@
 
 def _valfrommode(mode):
     try:
-        val = _modedict[mode]
+        return _modedict[mode]
     except KeyError:
-        if mode not in [0, 1, 2]:
-            raise ValueError("Acceptable mode flags are 'valid' (0),"
-                             " 'same' (1), or 'full' (2).")
-        val = mode
-    return val
+        raise ValueError("Acceptable mode flags are 'valid',"
+                         " 'same', or 'full'.")
 
 
 def _bvalfromboundary(boundary):
     try:
-        val = _boundarydict[boundary] << 2
+        return _boundarydict[boundary] << 2
     except KeyError:
-        if val not in [0, 1, 2]:
-            raise ValueError("Acceptable boundary flags are 'fill', 'wrap'"
-                             " (or 'circular'), \n  and 'symm'"
-                             " (or 'symmetric').")
-        val = boundary << 2
-    return val
+        raise ValueError("Acceptable boundary flags are 'fill', 'wrap'"
+                         " (or 'circular'), \n  and 'symm'"
```

Comment:
I think a manual line break is not needed here since most consoles are softwrapping messages anyways. A small nitpick I would suggest the following; first giving the actual full mode name and then the alias. 

```
raise ValueError("Unknown flag type given. Valid options are 'fill', 'wrap'"
                 " 'circular' (or 'circ' ), and 'symmetric' (or 'symm' ).")
```

Reply:
Ok I changed it similarly